src/resources/js/main.ts

```diff
@@ -75,27 +100,34 @@ function jsApiFunction() {
                         <td>${person.hair_color}</td>
                         <td>${person.eye_color}</td>
                         `;
-        table.append(tr);
+        table!.append(tr);
     }
 
-    function showLoader(nodeIdToAppend) {
+    function showLoader(nodeIdToAppend: string) {
         let node = document.getElementById(nodeIdToAppend);
         
         let loader = document.createElement('span');
         loader.id = nodeIdToAppend + '-loader';
         loader.className = 'loader';
-        node.append(loader);
+        node!.append(loader);
     }
 
-    function hideLoader(nodeId) {
+    function hideLoader(nodeId: string) {
         let loader = document.getElementById(nodeId + '-loader');
-        loader.remove();
+        loader!.remove();
```

Comment:
I've seen this before what is `!.`? Should we introduce it as a part of the course? I feel like optional chaining `?.` might be required.

Reply:
`tsc` was throwing an error here because `document.getElementById` could return `null`. The `!` allows me to assert that the value won't be `null` without putting more complicated logic in place.
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

This is actually my first time using it myself in Typescript, but I've used it quite a bit in C# before.

Reply:
This is amazing I've been using nullish coalescing. I like this soo much more
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing